cmd/wallet-web/src/router/index.js

```diff
@@ -4,17 +4,31 @@ Copyright SecureKey Technologies Inc. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0
 */
 
+import { computed } from 'vue';
 import { createRouter, createWebHistory } from 'vue-router';
 import store from '@/store';
+import { createKeyPair, gnapRequestAccess } from '@/mixins';
 import routes from './routes';
 
 const router = createRouter({
   history: createWebHistory(__webpack_public_path__),
   routes,
 });
 
-router.beforeEach((to, from, next) => {
+router.beforeEach(async (to, from, next) => {
   store.dispatch('agent/flushStore');
+  if (to.path === '/gnap') {
+    const gnapKeyPair = await createKeyPair();
```

Comment:
we creating key pair everytime someone access "*/gnap" in URL? this solution doesn't look scalable.
For example: you refresh this page and it creates key pair again?

Reply:
as we discussed offline we are going to finalize the creation of key pair on monday with the team

Reply:
Few lines below there is already TODO related to this added  @1699 it can be part of clientt persistance change which is already part of epic https://github.com/trustbloc/wallet/blob/53e281fab828cbd04a758d244970d0bbfe9b4b1c/cmd/wallet-web/src/router/index.js#L28